app/database_etl/test_adjustment_handler/test_adjustment.py

```diff
@@ -123,50 +149,34 @@ def pystan_adjust(self, model_params: Dict, execution_params: Dict = {}) -> Unio
         except ValidationError as err:
             print("Error: ", err.messages)
             return None, None, None
-
-        satisfactory_model_set_found = False
-        n_modelsets = 0
-
-        while not satisfactory_model_set_found:
-            satisfactory_model_set = []
-            for _ in range(self.n_replicates):
-                n_trials = 0
-                satisfactory_model_found = False
-
-                while not satisfactory_model_found:
-                    # Attempt to fit a model
-                    summary_df_parsed, satisfactory_model_found = self.fit_one_pystan_model(model_params)
-                    # If number of attempts exceeded, return Nones
-                    n_trials += 1
-                    if n_trials >= self.trials_lim:
-                        print('no models met the HMC diagnostics in {trials_lim} trials')
-                        return None, None, None
-
-                satisfactory_model_set.append(summary_df_parsed)
-
-            satisfactory_models_df = pd.DataFrame(satisfactory_model_set)
-
-            # see whether all results are within 10% of median; if not, rerun
-            model_medians = satisfactory_models_df['50%']
-            median = model_medians.median()
-            consistent_results_found = model_medians.between(0.9 * median, 1.1 * median).all()
-            median_run = satisfactory_models_df[satisfactory_models_df['50%'] == median].iloc[0]
-
-            lower, upper = arviz.hdi(median_run['samples'], credible_interval_size)
-            best_fit = median_run['fit']
-
-            try:
-                raw_prev = model_params['y_prev_obs'] / model_params['n_prev_obs']
-                bounded = result_is_bounded(median, raw_prev)
-            except (ZeroDivisionError, ValueError):
-                return None, None, None
-
-            satisfactory_model_set_found = consistent_results_found and bounded
-
-            n_modelsets += 1
-            if n_modelsets > self.modelsets_lim:
+        
+        
+        n_trials = 0
+        satisfactory_model_found = False
+
+        while not satisfactory_model_found:
+            # Attempt to fit a model
+            summary_df_parsed, satisfactory_model_found = self.fit_one_pystan_model(model_params)
+            # If number of attempts exceeded, return Nones
+            n_trials += 1
+            if n_trials >= self.trials_lim:
+                print('no models met the HMC diagnostics in {trials_lim} trials')
                 return None, None, None
+        
+        
+        satisfactory_model_set=summary_df_parsed
+            
+        satisfactory_models_df = pd.DataFrame(satisfactory_model_set)
+        median = satisfactory_models_df.at['50%','prev']
+        lower, upper = [satisfactory_models_df.at['2.5%','prev'],satisfactory_models_df.at['97.5%','prev']]
+        best_fit = satisfactory_models_df
 
+        try:
```

Comment:
Move try/except to be within loop that starts on line 157, after line 160

set `satisfactory_model_found = satisfactory_model_found & bounded`